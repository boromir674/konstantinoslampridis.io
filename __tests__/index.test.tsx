/**
 * @jest-environment jsdom
 */

import { jsx } from "@emotion/react";
import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom"; // Custom jest matchers to test the state of the DOM
import IndexPage from "../src/pages/index";

describe("Test App width = 700", () => {
  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <ResponsiveContext.Provider value={{ width: 700 }}>
        <IndexPage />
      </ResponsiveContext.Provider>
    );
    expect(mobile).toMatchSnapshot();
  });
});

describe("Test App width = 900", () => {
  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <ResponsiveContext.Provider value={{ width: 900 }}>
        <IndexPage />
      </ResponsiveContext.Provider>
    );
    expect(mobile).toMatchSnapshot();
  });
});
