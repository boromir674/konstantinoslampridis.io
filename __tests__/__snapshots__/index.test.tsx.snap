// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App width = 700 matches the snapshot 1`] = `
<div>
  <main
    style="color: rgb(35, 33, 41); padding: 96px; font-family: -apple-system, Roboto, sans-serif, serif; background-color: rgb(228, 136, 154);"
  >
    <div
      style="display: flex; flex-flow: row nowrap; cursor: pointer; width: 48px; user-select: none;"
    >
      <div
        style="padding-top: 0px; box-sizing: border-box; width: 100%; flex: 0 0 auto;"
      >
        <div
          style="width: 48px; height: 26px; border-radius: 16px; background: rgb(222, 222, 222); transition: all 100ms;"
        />
      </div>
      <div
        style="margin-left: -100%; box-sizing: border-box; width: 100%; flex: 0 0 auto;"
      >
        <div
          style="transition: all 100ms; padding-top: 4px; padding-left: 5px;"
        >
          <div
            style="width: 18px; height: 18px; size: 18px; border-radius: 16px; background-color: rgb(255, 255, 255); transition: all 100ms;"
          />
        </div>
      </div>
    </div>
    <div
      class="computerContainer"
    >
      <div
        class="Header"
      />
      <div
        class="Side-Profile"
      >
        <h2>
          Profile
        </h2>
      </div>
      <div
        class="Main-Pane"
      >
        <div
          class="Introduction"
        >
          <p>
            Hi, I am Konstantinos Lampridis :)
          </p>
        </div>
        <div>
          <h3>
            Open Source Portfolio
          </h3>
        </div>
        <div
          class="Career"
        >
          <h3>
            Professional Career
          </h3>
        </div>
      </div>
      <div
        class="Footer"
      />
    </div>
    <div
      class="css-1l2b09b"
    >
      some text
    </div>
    <button
      class="css-1ap15ok"
      data-testid="button-id"
    >
      This my button component.
    </button>
    <p
      class="tempp"
      css="[object Object]"
      data-testid="dynamic-el"
    >
      Test text!
    </p>
  </main>
</div>
`;

exports[`Test App width = 900 matches the snapshot 1`] = `
<div>
  <main
    style="color: rgb(35, 33, 41); padding: 96px; font-family: -apple-system, Roboto, sans-serif, serif; background-color: rgb(228, 136, 154);"
  >
    <div
      style="display: flex; flex-flow: row nowrap; cursor: pointer; width: 48px; user-select: none;"
    >
      <div
        style="padding-top: 0px; box-sizing: border-box; width: 100%; flex: 0 0 auto;"
      >
        <div
          style="width: 48px; height: 26px; border-radius: 16px; background: rgb(222, 222, 222); transition: all 100ms;"
        />
      </div>
      <div
        style="margin-left: -100%; box-sizing: border-box; width: 100%; flex: 0 0 auto;"
      >
        <div
          style="transition: all 100ms; padding-top: 4px; padding-left: 5px;"
        >
          <div
            style="width: 18px; height: 18px; size: 18px; border-radius: 16px; background-color: rgb(255, 255, 255); transition: all 100ms;"
          />
        </div>
      </div>
    </div>
    <div
      class="computerContainer"
    >
      <div
        class="Header"
      />
      <div
        class="Side-Profile"
      >
        <h2>
          Profile
        </h2>
      </div>
      <div
        class="Main-Pane"
      >
        <div
          class="Introduction"
        >
          <p>
            Hi, I am Konstantinos Lampridis :)
          </p>
        </div>
        <div>
          <h3>
            Open Source Portfolio
          </h3>
        </div>
        <div
          class="Career"
        >
          <h3>
            Professional Career
          </h3>
        </div>
      </div>
      <div
        class="Footer"
      />
    </div>
    <div
      class="css-1l2b09b"
    >
      some text
    </div>
    <button
      class="css-1ap15ok"
      data-testid="button-id"
    >
      This my button component.
    </button>
    <p
      class="tempp"
      css="[object Object]"
      data-testid="dynamic-el"
    >
      Test text!
    </p>
  </main>
</div>
`;
